### PR TITLE
feat: Add benchmarking for encryption/decryption

### PR DESCRIPTION
**Pull request by Anupam-1208**
original issue: [https://github.com/juspay/hyperswitch/issues/6228]

Added benchmarking for Single and Batch data for variable input sizes

Features:
- Single data of variable sizes 
- Batch data of variable number of Input for fixed size data
- Input size can be adjusted before benchmarking 

Sample Benchmark:
Screenshot 2024-11-04 at 11 45 58 AM

### CODE REVIEW

File: benches/encryption_bench.rs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+const SINGLE_BENCH_ITERATION: u32 = 10;
+const BATCH_BENCH_ITERATION: u32 = 10;
+
+criterion_main!(benches);
+criterion_group!(benches, criterion_data_encryption_decryption, criterion_batch_data_encryption_decryption);
+
+pub fn criterion_data_encryption_decryption(c: &mut Criterion) {
+    let rt = Runtime::new().unwrap();
+    let custodian = Custodian::new(Some(("key".to_string(),"value".to_string())));
+    let config = config::Config::with_config_path(config::Environment::which(), None);
```
Comment (dracarys18) [inline benches/encryption_bench.rs:23]: You can directly use Environment::Dev

File: benches/encryption_bench.rs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    let state = rt.block_on(async {
+         let state = AppState::from_config(config).await;
+         return state;
+    });
+    let identifier = Identifier::User(String::from("user_12345"));
+
+    {
+        let mut group = c.benchmark_group("data-encryption-single");
```
Comment (dracarys18) [inline benches/encryption_bench.rs:31]: Can you create the key `user_12345` in the DB before running the bench

**Review by dracarys18 — CHANGES_REQUESTED** [review]

**Comment by dracarys18** [comment]
Hey @Anupam-1208 Thanks for the PR! I am getting the following error during my first ever run, Likely due to my second comment, Please take a look

```
Failed while encrypting: Failed to Get the data key
├╴at src/errors/crypto.rs:40:32
│
╰─▶ Failed to get the value from the database
    ╰╴at src/errors/db.rs:45:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

File: benches/encryption_bench.rs (hunk 2 above)
Comment (Anupam-1208) [reply benches/encryption_bench.rs:31]: Yes, I created a new DataKey using the following api and ran the benchmark, will update the description
```
curl --location 'localhost:5000/key/create' \
--header 'Content-type: application/json' \
--data '{
  "data_identifier": "User",
  "key_identifier": "user_12345"
}'
```

Comment (Anupam-1208) [reply benches/encryption_bench.rs:31]: Screenshot 2024-11-04 at 11 45 58 AM

File: benches/encryption_bench.rs (hunk 1 above)
Comment (Anupam-1208) [reply benches/encryption_bench.rs:23]: Sure,cWill update code

Comment (Anupam-1208) [reply benches/encryption_bench.rs]: Updated, please check the code, also create Key using the curl below

File: benches/encryption_bench.rs (hunk 2 above)
Comment (Anupam-1208) [reply benches/encryption_bench.rs:49]: @dracarys18 added the logic to create user at the time of running benchmark. Can you review it again?

**Review by dracarys18 — APPROVED** [review]
LGTM!

**Comment by dracarys18** [comment]
@Anupam-1208 Please fix formatting and clippy errors

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]
@Anupam-1208 Please fix failing CI checks.

**Comment by Anupam-1208** [comment]
@dracarys18 @ArjunKarthik added formatting changes

**Review by dracarys18 — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]
LGTM

**Comment by dracarys18** [comment]
Thanks for the contribution @Anupam-1208

**Pull request merged**